client/state/themes/utils.js

```diff
@@ -2,25 +2,47 @@
  * External dependencies
  */
 import startsWith from 'lodash/startsWith';
-import { filter, get, map, mapKeys, omit, omitBy, split } from 'lodash';
+import {
+	every,
+	filter,
+	get,
+	includes,
+	map,
+	mapKeys,
+	omit,
+	omitBy,
+	some,
+	split
+} from 'lodash';
 
 /**
  * Internal dependencies
  */
-import { isThemeMatchingQuery } from 'lib/query-manager/theme/util';
 import { DEFAULT_THEME_QUERY } from './constants';
 
 /**
  * Constants
  */
 const REGEXP_SERIALIZED_QUERY = /^(?:(\d+):)?(.*)$/;
+const SEARCH_TAXONOMIES = [ 'subject', 'feature', 'color', 'style', 'column', 'layout' ];
```

Comment:
> Wait! This is just for testing, right? Phew.

No, not just for testing. This is used by `isThemeMatchingQuery`, which up until this PR would've been used as  `ThemeQueryManager`'s `matches()` method. Since we're dropping the latter now, only one use remains: client-side filtering of themes obtained from a Jetpack site's endpoint.

We do this [here](https://github.com/Automattic/wp-calypso/blob/a87e24abe8669dd5da2a75220169d5a91849dde7/client/state/themes/actions.js#L227), and it essentially supersedes [this stuff](https://github.com/Automattic/wp-calypso/blob/a87e24abe8669dd5da2a75220169d5a91849dde7/client/components/data/themes-list-fetcher/index.jsx#L115-L138).

> I'm worried about this though, because filtering should be completely transparent to the theme state.

I'd agree in principle, but as long as the JP endpoint differs from the WPcom one in this regard, I'd filter as early as possible after retrieval _from_ the endpoint to be transparent afterwards. 

I can pull the taxonomies from `theme-filters`. I think the themes list endpoint doesn't return them anyway (only the details one does), and the Jetpack one only returns `feature` (in the guise of `tags`, but we normalize that to `feature`).

Bottom line is that now that we don't use this for TQM, we might actually drop most of the taxonomies except for `feature` since they're not present in their remaining use case -- JP results filtering -- anyway.